src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -241,7 +241,6 @@ private IAsyncResult BeginSendFileInternal(string fileName, byte[] preBuffer, by
             FileStream fileStream = OpenFile(fileName);
 
             TransmitFileAsyncResult asyncResult = new TransmitFileAsyncResult(this, state, callback);
-            asyncResult.StartPostingAsyncOp(false);
```

Comment:
Here's what I believe happened.  Would be nice if @stephentoub or someone else could confirm.

When you construct an Overlapped, you can either use Pack or UnsafePack.  Pack flows the EC (which includes current identity, etc).  UnsafePack does not.

Flowing the EC used to be expensive, because you had to make a copy of it.  So, we tried to avoid it when possible.  We used UnsafePack and cached ECs using ContextAwareResult, and invoked EC.Run ourselves.

When Sockets got ported to Core, UnsafePack didn't exist, because now EC is immutable and flowing it is cheap.  So, all the UnsafePack calls got changed to Pack, and we ended up flowing EC twice.

Now, we should be able to remove ContextAwareResult without breaking EC flow and without hurting performance.